include/zephyr/bluetooth/addr.h

```diff
@@ -81,6 +81,13 @@ static inline int bt_addr_cmp(const bt_addr_t *a, const bt_addr_t *b)
 	return memcmp(a, b, sizeof(*a));
 }
 
+/** @brief Compare Bluetooth LE device addresses for equality.
+ */
+static inline bool bt_addr_le_eq(const bt_addr_le_t *a, const bt_addr_le_t *b)
+{
+	return ((a->type == b->type) && (memcmp(a->a.val, b->a.val, sizeof(a->a.val)) == 0));
```

Comment:
Why not just `return (bt_addr_le_cmp(a, b) == 0);`?

Reply:
I suppose it can be. But I assumed we would like to deprecate `bt_addr_le_cmp`?

Also `bt_addr_le_cmp` is not well defined. The documentation just says "compare" and uses (>)/(<) operators as if it already had a defined meaning on `bt_addr_le_t`. The implementation also just `memcmp`s the whole structure, which is not packed, so probably not meant to be accessed as a string?

Reply:
The main reason we've had "cmd" functions is to potentially allow them to be used with sorting functions that take a callback with a similar signature (akin to memcmp() and strcmp()). I suppose use cases where you need to sort addresses are quite rare, so we could indeed consider deprecating the cmp function. However if that's part of the intention with these new APIs, please include that information in the commit message. Answering the question "why?" is usually the most important task of the commit message.

Reply:
Ok. I think I was too quick with assuming we want to deprecate `cmp`. I see `cmp` may indeed be useful for algorithms like heaps and sorted log(n) lookups. In that case it does not even matter how the addresses are sorted as long as the transitive (<=) property holds.

I will update this PR with the definition of `bt_addr_le_eq` that you propose.

I will leave the discussion about whether `bt_addr_le_cmp` is well defined for the future. Just as food for thought: My current thinking is that it's best if we follow the rules of C and code for the C abstract machine. So it's not good that `bt_addr_le_cmp` is just a `memcmp`. I believe the compiler is in principle allowed to insert padding as it pleases after any member in a struct. The fully compliant definition of `bt_addr_le_cmp` should be `return (b.type - a.type) || bt_addr_cmp(&a.a, &b.a)`, and `bt_addr_cmp` should be `memcmp(a->val, b->val, sizeof(a->val));`.